mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala

```diff
@@ -168,9 +169,19 @@ class CrossValidator @Since("1.2.0") (@Since("1.4.0") override val uid: String)
       val validationDataset = sparkSession.createDataFrame(validation, schemaWithoutFold).cache()
       instr.logDebug(s"Train split $splitIndex with multiple sets of parameters.")
 
+      val sparkContext = sparkSession.sparkContext
+      val oldJobGroup = sparkContext.getLocalProperty(SparkContext.SPARK_JOB_GROUP_ID)
+      val cvJobGroup = s"${this.uid}_job_group"
+      sparkContext.setLocalProperty(SparkContext.SPARK_JOB_GROUP_ID, cvJobGroup)
```

Comment:
Hm, wouldn't this unset previous job group. If users want to cancel `oldJobGroup` group, looks like it's not going to work

Reply:
Yes this is a corner case. I don't have good idea.
If we support cancelling by job group prefix, then it could address this issue.
CC @srowen Any thoughts ?